### DISCUSSION TITLE
Browsing a source member

### DISCUSSION THREAD

**Discussion by Gregory1701** [codefori/vscode-ibmi#1868] [IBM i]
Hi, 
So this is the scenario, a colleague is editing a source member in his library. I open my filter which is set up to look at his library source and set to protected. Then I find the source and right click on it, go to the open menu and click browse. I get this: 
![image](https://github.com/codefori/vscode-ibmi/assets/132368707/85b08db1-3444-4361-bc8d-7c1d060907b8)

I am not authorized to his library, by design, and my filter is set to protected, not necessary, but again by design. Shouldn't I be able to browse a member that the owner has open in edit mode?

**Comment by sebjulliand** [comment]
Hi @Gregory1701 ,
the extension itself doesn't prevent you from opening the source member; CPYTOSTMF does. It's called under the hood to retrieve the source content and when a member is opened for edition in SEU or RDi, that happens. The fact your filter is protected or not has no incidence on this.
We would need to find another way to get the source content, regardless of its allocation, to fix this I think 🤔.

**Reply by Gregory1701** [reply]
Do you or anyone else know how RDi retrieves a source for browse and is able to do this when someone else is also viewing/editing the same source member?

**Reply by chrjorgensen** [reply]
Hi @Gregory1701 
RDi does not use CPYTOSTMF to retrieve a source member - as Code for IBM i does... we're currently exploring alternatives to the CPYTOSTMF command and may eventually fix this issue. No fixed target for this, though...

**Reply by sebjulliand** [reply]
I'm pretty sure it uses record access level to read the member, which may not be impacted by the lock put on when it's opened for editing by RDi or SEU.
But, we are looking at other means of reading the member, not using CPYTOSTMF. That may eventually help with this issue 😉

**Comment by jkyeung** [comment]
Is there a reason you can't just query the source directly? CPYTOSTMF seems incredibly roundabout to me.

I've built my own set of download/transfer/comparison tools which are centered around

```sql
create alias qtemp.foo for foolib.foofil(foombr);
select * from qtemp.foo;
```

For uploading into SRC-PFs, I use INSERT. It all seems to work pretty well for me.
